Docs/14 Classes and Functions/README.md

```diff
@@ -0,0 +1,131 @@
+
+```fsharp
+
+let staticChild name (currentName: string cval) =
+    button {
+        // trigger currentName changes
+        onclick(fun _ -> currentname.Publish name)
+        $"Click {name}!"
+    }
+
+let staticParent () =
+    let currentName = cval("Nobody")
+    fragment {
+        // doesn't re-render
+        h1 { $"Initial Name: {currentName.Value}" }
+
+        // render on currentName changes
+        adaptiview() {
+            let! currentName = currentName
+            h1 { $"Hello, {currentName}!" }
+        }
+
+        staticChild "Peter" currentName
+        staticChild "Frank" currentName
+    }
+```
+
+This reduces the need to have classes for most of the codebase, you can combine dynamic data with Fun.Blazor's DSL to go as functional as you need.
+
+### Benefits and drawbacks
+
+The benefits are:
+
+- Simpler to test
+- Easier to reason about
+- Composable
+
+They are often the preferred way to go for most F# users.
+
+The downsides are:
+
+- Don't have access to aspnet's default DI container
+- Don't play well raw blazor
+- They need to be wrapped in components in order for C# consumers to use
+
+In short the closer you want to get to Blazor, the least functions work nicely with it.
+
+However, most of the time a function is the simple yet correct choice.
+
+## Classes
+
+Blazor by default re-renders in certain conditions like after an event handler is executed, changing parameters/cascading parameters and when calling `this.StateHasChanged()` so it plays well with mutable values even in F#.
+
+A simple counter example would be the following:
+
+```fsharp
+type Counter() =
+    inherit FunComponent()
+
+    let mutable counter = 0
+    [<Inject>]
+    member val Logger: ILogger<Counter> = Unchecked.defaultof<_>
+
+    override this.Render() =
+        fragment {
+            h1 { $"Count: {counter}" }
+
+            button {
+                onclick (fun _ ->
+                    let newCount = counter + 1
+                    Logger.LogDebug("Old Value: {count} - New Value: {newCount}", counter, newCount)
+                    counter <- newCount
+                )
+                "Increment"
+            }
+        }
+```
+
+In this case we didn't add any dynamic values like Adaptive data, Stores or Observables and our counter is re-rendered and works as expected, blazor applies a diffing algorithm to know that it only needs to re-render the `h1` tag. you can also however use dynamic data if required.
+
+### Benefits and drawbacks
+
+The benefits are:
+
+- Offer full decorator support `Parameter`, `Cascading Parameter`, `Route`, `Inject`
+- Has direct access to the DI container in aspnet
+- Consumption is transparent for C# consumers
+- Plays well with the blazor framework under the hood.
+
+The drawbacks are:
+
+- Re-rendering requires diffing which can be expensive if you're not using dynamic data
```

Comment:
> Regarding for performance, both function or component style requires diffing. Because function style node is just a delegate which is waiting to be called by its parent.

What I tried to mean here is that (at least I think, let me know if I'm wrong here) that a bigger tree diff is more costly than diffing just what the adaptiview has to diff

but I'll make the distinction between FunComponent and FunBlazorComponent for those cases.


> But if we use FunBlazorComponent which the event trigger rerender is disabled by default, so it can be used together with adaptive or observable and only specific area of the component's content will need to be used to computate for diff when data is changed:

Ah yes that's what I wanted to say, I missed the FunBlazorComponent entirely, I will change this section, thanks!